### DISCUSSION TITLE
As of 4.0.4, what features do you want to be implemented the most?

### DISCUSSION THREAD

**Discussion by Pentaphon** [transmission/transmission#6084] [General]
I have to admit, I am totally surprised that Transmission cannot tell you how long a torrent has been seeding. https://github.com/transmission/transmission/issues/870

What about the community? We should start prioritizing what features we need the most so contributors can prioritize their PRs.

**Comment by AK220011** [comment]
Binding to an interface just like qbittorrent which makes it very easy to bind to a vpn client.

EDIT: For Qt on windows.

**Reply by wally-crunk** [reply]
This, by far, the most of all the suggestions posted to date.

**Reply by TejSwift** [reply]
Bump DEC 2025

**Comment by bayarnd** [comment]
#1 missing feature for me: unlike other torrent clients I have used, Transmission does not open a window showing the content of multi file torrents when I add the magnet to the list, allowing me to select/deselect which files from the torrent I want to download. it appears I have to start the download of the complete package before I can open the info and deselect all torrent files I did not want.

**Comment by valcomm** [comment]
> Transmission does not open a window showing the content of multi file torrents when I add the magnet to the list, allowing me to select/deselect which files from the torrent I want to download. it appears I have to start the download of the complete package before I can open the info and deselect all torrent files I did not want.

I'll join to this issue.  It is very inconvenient in case where only few files are needed from the magnet link.  
It can be an additional command line option - pause after metadata downloaded (default) or download content after getting the metadata.

Also it can be a new hook - for the event "after metadata downloaded" (skipped completely for .torrent files).

Or, perfectly it can be the both features above

For the GUI client it might be messy.  For example QBittorrent allows adding multiple magnets and each gets a modal window for the coming content list.  Though there is a time limit for getting the metadata.  After it reaches the limit magnet should be added again.  This approach isn't ideal, but works for the most cases.

**Reply by killemov** [reply]
Yes for pausing after metadata complete, no for waiting and then showing a dialog.

**Comment by NiklasBr** [comment]
Reverse availability downloading: Make it a higher importance to download less-available pieces than pieces of a torrent shared my many peers. For example consider this simplified availability chart of a file:

|                 |   Peer A    |    Peer B    |    Peer   C     |
|-----------|------------|-----------|--------------|
|   Piece 1   |        X       |        X        |        X          |
|   Piece 2   |        X       |                 |        X          |
|   Piece 3   |                |                 |        X          |
|   Piece 4   |                |         X       |        X          |

Transmission should be asking for Piece 3 from Peer C as early and often as possible

**Reply by Pentaphon** [reply]
>Reverse availability downloading: 

@NiklasBr Have you created an issue for this? I looked and there's nothing like this on the issues page.

**Reply by NiklasBr** [reply]
I read about it on the old Trac so I assumed it had been moved here.

**Reply by Pentaphon** [reply]
You should make a new issue for it here on Github then.

**Reply by andreygursky** [reply]
It turns out, this feature was implemented long ago in 2.30b1 https://github.com/transmission/transmission/commit/5dffc2814d557f197aaeb83375849c163e1851f7 (17.02.2011). But removed in 4.0.0-beta.1 https://github.com/transmission/transmission/pull/2021 (25.10.2021).

**Reply by NiklasBr** [reply]
Too bad it was removed, I am confident that it would be a net-positive for the torrent "ecosystem" as a whole and worth a bit of extra CPU usage locally.

**Reply by killemov** [reply]
Weird that priority of rarity of pieces was removed as it is one of the base principles of the Bittorrent protocol. Perhaps this happened because of the implementation of sequential downloading. If the number of seeders are high enough this should not be a problem though. So switching to the rarity strategy below X (=16?) seeders should comprise the best of both worlds.

**Comment by likeaross** [comment]
I for one would love to see peer, speed and ratio limits on a per tracker basis instead of just an individual torrent or global configuration.

**Comment by lostname420** [comment]
Add additional columns in the GUI and make them all sortable by clicking column header. Save path, % Complete, ETA, Tracker, Size, State, and Label as suggestions. Also make columns optional in settings so users aren't forced to change. This would change the sort options in the view menu. This could also be achieved by adding a "Detailed" or "Advanced" view.

This is the main functionality I've missed since switching to Transmission.

**Reply by GaryElshaw** [reply]
I'd like to be able to move columns.

**Reply by killemov** [reply]
Some windowing systems have this (almost) built in ... for tables.

**Comment by ajarora** [comment]
Sequential downloading by far.

Stoked for #597 and have a feeling it'll have a big impact. Thanks for all the work @pldubouilh, @rnhmjoj, @ckerr et al 🙏

**Comment by promac** [comment]
it would be great to have • the active filter split into 2 with **active seeding and active downloading** AND • the possibilities to select/deselect which files from the torrent you want to download.AND • to see in list view , the **comment**  after the Name of the torrent + the **torrent path** AND •  inside the Menu list "**SORT Transfer BY**" add ratio, uploaded, downloaded, nb peers, UL speed, comment, Creation Date,  transfer priority, groups) AND • **search by comment**. AND • Show torrent file (.torrent) in Finder. Many thanks and Congrats for all the work

**Comment by dantearmok** [comment]
I'd like to see two improvements to the Inspector window.

1) A stable/smarter sort for the UL and DL columns.  eg: when you sort by DL, the peers that both DL and UL should be shown first, then the DL only peers, then the UL only peers, and finally, at the end of the list, the peers that are just hanging out doin nuthin.

2) A new column that contains the tooltip information for each peer.  Currently, that information is all but unreadable because the 1s refresh of the list causes the peer to jump away.  Putting that information in a sortable list would be quite helpful.

**Comment by Yohmi** [comment]
Hi all !

There's a function, which I think is missing (unless I'm missing something !) is to have the ability to move files location to another volume without altering the other parts of the path (so I can keep the subfolder structures where all the torrents are properly sorted in custom folder names).
A user asked for this on the support forum here : https://forum.transmissionbt.com/viewtopic.php?p=81178
It's not the first time I stumbled upon this issue, and back then I painstakingly relocated all torrents one by one after copying all files to the new destination with paused torrents. It was not a lot of fun. And now my file count is much higher, I don't see myself doing this again. 

Theres another one which could be interesting to me (but really not a major thing), is to be able to combine several torrents pointing to the same file, for cross-seed purpose with trackers using passkeys.

thanks !

**Reply by killemov** [reply]
Currently moving content across filesystem barriers is quite resource intensive, sometimes even leading to servercide. So the not fun way of doing it is the way I would advise anyway. Maybe this should be implemented in the same vein verifying is implemented. Also a set of predetermined paths would be welcome.

**Comment by hatyi** [comment]
I'm using the app in a Docker container (4.0.4) with the web ui. Being able to select which file I want to download would be awesome.

**Reply by Pentaphon** [reply]
>Being able to select which file I want to download would be awesome.

That's a limitation of the official WebUI. I would try using https://github.com/openscopeproject/TrguiNG if you want to do this now.

**Reply by hatyi** [reply]
Thanks so much for letting me know about this.

**Reply by killemov** [reply]
@hatyi If you want to do WAY more with the web-UI you also might want to check out [Shift](https://github.com/killemov/Shift).

**Comment by BalticViking** [comment]
Regarding the latest update Transmission 4.1.0 (272401184f)-- GROUPS changes 

Just want to share that I personally really prefer having each torrent display it's color "Group" DOT identifier. I have a huge number of torrents organized by groups.  

My new update places one DOT at the top of each Group. Super challenging to do a quick scroll through 100's to a particular group. I really prefer each line having a group dot identifier.

Did a quick run through settings but didn't find an option to do that-- If there is one... would love some assistance to help me activate that. :-)

Thanks all!
BV
